### PR TITLE
Add metrics for Monitor Promo

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to track view and submit events for the new `ProductPromo` componet

## This pull request

- Adds new Glean events

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-10212

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Screenshot 2024-08-08 at 12 15 09 PM
Screenshot 2024-08-08 at 12 14 52 PM

### CODE REVIEW

File: packages/fxa-settings/src/constants/index.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+
```
Comment (LZoog) [inline packages/fxa-settings/src/constants/index.tsx:5]: Could be worth removing from the diff if you need to push anyway 🤷‍♀️

File: packages/fxa-settings/src/lib/glean/index.test.ts [hunk 2]
```diff
@@ -725,6 +725,30 @@ describe('lib/glean', () => {
...
+        GleanMetrics.accountPref.promoMonitorSubmit();
+        const spy = sandbox.spy(accountPref.promoMonitorSubmit, 'record');
+        await GleanMetrics.isDone();
+        sinon.assert.calledOnce(setEventNameStub);
+        sinon.assert.calledWith(
+          setEventNameStub,
+          'account_pref_promo_monitor_submit'
+        );
+        sinon.assert.called(spy);
+      });
```
Comment (LZoog) [inline packages/fxa-settings/src/lib/glean/index.test.ts:751]: Can we add a couple tests in `ProductPromo/index.test.tsx` too to make sure that 'reason' is as expected? 🙏

**Review by LZoog — APPROVED** [review]
Would love to have those couple other tests but LGTM. Not sure why the frontend integration tests are failing at first glance but I'm sure it's an easy fix. Thanks!

**Comment by vbudhram** [comment]
Merging this in, looks like failure posting status to JIRA. Maybe something changed with the user?

```
"clouserw"
jq: error (at <stdin>:406): null (null) cannot be matched, as it is not a string

Exited with code exit status 5
```

**Pull request merged**